assets/systemd-unit-upgrader/teleport-upgrade

```diff
@@ -0,0 +1,299 @@
+#!/bin/bash
+
+set -euo pipefail
+
+
+conf_dir="${TELEPORT_UPGRADE_CONFIG:-/etc/teleport-upgrade.d}"
+
+# get_conf is a helper for loading values from the configuration dir. supports very
+# rudimentary comment stripping in order to let us include basic comments in generated
+# configs.
+get_conf() {
+    conf_name=$1
+    default_value=$2
+    
+    # if file does not exist, use default
+    if ! test -f $conf_dir/$conf_name; then
+        echo "$default_value"
+        return 0
+    fi
+
+    # strip trivially obvious comments, empty lines, and leading/trailing whitespace.
+    normalized_value="$(grep -v -e '^# ' -e '^[[:space:]]*$' <"$conf_dir/$conf_name" | awk '{$1=$1};1')"
+
+    # if variable is empty post-normalization, use default
+    if [[ -z "$normalized_value" ]]; then
+        echo "$default_value"
+        return 0
+    fi
+
+    echo "$normalized_value"
+}
+
+
+# upgrade_endpoint is the base host+path used to load the /version and /critical endpoints
+upgrade_endpoint="$(get_conf endpoint update.gravitational.io/v1/stable/cloud)"
```

Comment:
```suggestion
upgrade_endpoint="$(get_conf endpoint updates.releases.teleport.dev/v1/stable/cloud)"
```

Note that I'm going to file a PR for a dev bucket at `updates.releases.development.teleport.dev` as well